dashboard/src/pages/[workspaceSlug]/[appSlug]/logs/index.tsx

```diff
@@ -56,33 +57,41 @@ export default function LogsPage() {
           from: fromDate,
         },
         updateQuery: (prev, { subscriptionData }) => {
+          // if there is no new data, just return the previous data
           if (!subscriptionData.data) {
             return prev;
           }
 
+          const prevLogs = prev.logs;
+
           // if there are no previous logs just return the new ones
-          if (!prev.logs || prev.logs.length === 0) {
+          if (!prevLogs || prevLogs.length === 0) {
             return subscriptionData.data;
           }
 
-          const previousLogsLength = prev.logs.length;
-          const newLogsLength = subscriptionData.data.logs.length;
-          const timestampOfFirstPreviousLog = prev.logs[0].timestamp;
-          const timestampOfFirstSubscriptionLog =
-            subscriptionData.data.logs[newLogsLength - 1].timestamp;
-          const sameLogs =
-            previousLogsLength === newLogsLength &&
-            timestampOfFirstPreviousLog === timestampOfFirstSubscriptionLog;
+          const newLogs = subscriptionData.data.logs;
 
-          if (sameLogs) {
-            return subscriptionData.data;
+          // Next, we need to understand if the new logs are the same as the previous ones.
+          // We'll first check if the length of the logs is the same.
+          // We'll then pick the first log from `prev` and see if we can find it in `subscriptionData.data`.
+          // If it exists, we'll assume that the logs are the same and we'll return `prev`.
+          // NOTE: We can't compare elements in the array becayse they are sent out of order. The logs are sorted by timestamp in the LogsBody component.
+
+          const prevAndNewLogsAreTheSame =
+            prevLogs.length === newLogs.length &&
+            newLogs.some(
+              (log) =>
+                log.timestamp === prevLogs[0].timestamp &&
+                log.service === prevLogs[0].service,
+            );
+
+          if (prevAndNewLogsAreTheSame) {
+            return prev;
           }
 
-          const newLogs = subscriptionData.data.logs;
-
+          // if the logs are not the same, it means we got new logs. We'll merge the new logs with the existing logs.
```

Comment:
```suggestion
          // if the logs are not the same, it means we've got new logs. We'll merge the new logs with the existing logs.
```

Also we've this format for multiline comments at most other files:

```
 /**
   * 
   * 
   */
```
which I prefer but I don't quite know if it makes sense for these kinds of lines (as opposed to functions, hooks, etc...)

Reply:
in this case, I think `//` makes sense. For hooks, etc. it makes sense to use the `/** ... */` syntax because the comment is then used by the IDE as you mentioned before